src/arith/iter_affine_map.cc

```diff
@@ -1062,58 +1099,59 @@ PaddedIterMapResult DetectPaddedIterMap(const Array<PrimExpr>& indices,
       [](const IterConstraint& a, const IterConstraint& b) { return a.expr_size < b.expr_size; });
 
   IterMapRewriter rewriter(analyzer, constrained_input_iters, simplify_trivial_iterators,
-                           &result.errors);
+                           &result->errors);
   // Step0.0: rewrite constraints in the order from size-small ones to size-big ones
   for (const IterConstraint& constraint : constraints) {
     auto res = rewriter.RewriteIterConstraint(constraint.iter, constraint.lower_bound,
                                               constraint.upper_bound);
-    if (result.errors.size()) {
-      return result;
+    if (result->errors.size()) {
+      return result_obj;
     }
   }
   if (!rewriter.CheckConstraints()) {
-    result.errors.push_back("Invalid constraints.");
-    return result;
+    result->errors.push_back("Invalid constraints.");
+    return result_obj;
   }
 
   // Step0.1: Check each index to determine required padding
-  bool allow_padding = !require_bijective;
+  bool allow_padding = check_level != IterMapLevel::Bijective;
```

Comment:
That is the claim~  I try to change padding to iter mark itself.

For example,`(x + 7)` `x` in [0, 8) => `IterMark(IterSplit(IterSum({x}, 7), lower_factor=1, extent=16, scale=1), extent=16` with left_pad=7, right_pad=1

Then `(x + 7) // 8` is mapped to range [0, extent//2) == [0, 2), though we have padding into iter mark, the IterSplit's range can be achieved when we only iterate `x` in it's original domain: (0 + 7) // 8 = 0, (7 + 7) // 8 = 1

Reply:
Good point, and that does maintain surjectivity for a single index.  I'm not entirely sure for the case of two indices, though.  For the same `x ∈ [0,8)`, the indices `[(x+7)//8, (x+7)%8]` would have the same padding `left_pad=7` and `right_pad=1`.  Even though each individual index can take any value in the output (`(x+7)//8 ∈[0,2)` and `(x+7)%8 ∈ [0,8)`), there are some coordinate pairs that cannot be generated for any value of `x` (e.g. `[0,0]` and `[1,7]`).

Reply:
I agree! This is where we should be careful. In `CheckMapping` with surjective mode when padding exists, we check `padded // LCM` and `padded % LCM`(or it's sub-splits) must not both exists. The case below depict this check:
```python
sum = 80 + y
dom_map = var_dom([(y, 176)])

# (80 + y) // 32 itself could be surjective
assert_iter_sum_pattern(
    {fld(sum, 32): (6, 2, 1)},
    dom_map,
 )

# (80 + y) % 2, ((80 + y) // 2) % 16) could be surjective, since they can be seen as sub-splits of (80 + y) % 32
assert_iter_sum_pattern(
    {flm(fld(sum, 2), 16): (16, 0, 1), flm(sum, 2): (2, 0, 1)},
    dom_map,
)

# but (80 + y) // 32, (80 + y) % 32 are not surjective
assert_iter_sum_failure({fld(sum, 32), flm(sum, 32)}, dom_map)
```

Other kinds of negatives like `(80 + y) // 32, (80 + y) // 4` would be banned by existing checking rule.